api/proxy/cmd/server/entrypoint.go

```diff
@@ -43,6 +44,34 @@ func StartProxyService(cliCtx *cli.Context) error {
 	if err := cfg.Check(); err != nil {
 		return err
 	}
+
+	// Initialize OpenTelemetry tracing
+	telemetryCfg := telemetry.ReadConfig(cliCtx)
+	if err := telemetryCfg.Verify(); err != nil {
+		return fmt.Errorf("invalid telemetry config: %w", err)
+	}
+
+	ctx, ctxCancel := context.WithCancel(cliCtx.Context)
+	defer ctxCancel()
+
+	shutdownTracer, err := telemetry.InitTracer(ctx, telemetryCfg)
+	if err != nil {
+		return fmt.Errorf("initialize tracer: %w", err)
+	}
+	defer func() {
+		if err := shutdownTracer(context.Background()); err != nil {
```

Comment:
Any reason to build a new context object here?

Reply:
I don't think so https://github.com/Layr-Labs/eigenda/pull/2204/commits/b7ab7cb8ec661b0c3cefc32d4fb4b482984ee4c3